pkg/migrations/sql/idempotent/base.sql

```diff
@@ -2350,6 +2358,239 @@ SET search_path = pg_temp;
+                )
+                WHERE ht.schema_name = schema_name_in
+            ),
+            up AS MATERIALIZED (
+                -- list of nodes we care about and whether they are up
+                SELECT
+                    x.node_name,
+                    _timescaledb_internal.ping_data_node(x.node_name) AS node_up
+                FROM (
+                    SELECT DISTINCT dht.node_name -- only ping each node once
+                    FROM dht
+                ) x
+            )
+            SELECT
+                dht.table_name as hypertable_name,
+                dht.node_name,
+                up.node_up
+            FROM dht
+            JOIN up ON (dht.node_name = up.node_name)
+        $function$
+        LANGUAGE sql
+        STRICT STABLE 
+        SECURITY DEFINER
+        --search path must be set for security definer
+        SET search_path = pg_temp;
+        REVOKE ALL ON FUNCTION SCHEMA_CATALOG.hypertable_node_up(name) FROM PUBLIC;
+        GRANT EXECUTE ON FUNCTION SCHEMA_CATALOG.hypertable_node_up(name) to prom_reader;
+    END IF;
+END;
+$block$
+;
+
+DO $block$
+BEGIN
+    IF SCHEMA_CATALOG.is_timescaledb_installed() AND SCHEMA_CATALOG.get_timescale_major_version() >= 2 THEN
+        CREATE OR REPLACE FUNCTION SCHEMA_CATALOG.hypertable_remote_size(schema_name_in name)
+        RETURNS TABLE(hypertable_name name, table_bytes bigint, index_bytes bigint, toast_bytes bigint, total_bytes bigint)
+        AS $function$
+            SELECT
+                dht.hypertable_name,
+                sum(x.table_bytes)::bigint AS table_bytes,
+                sum(x.index_bytes)::bigint AS index_bytes,
+                sum(x.toast_bytes)::bigint AS toast_bytes,
+                sum(x.total_bytes)::bigint AS total_bytes
+            FROM SCHEMA_CATALOG.hypertable_node_up(schema_name_in) dht
+            LEFT OUTER JOIN LATERAL _timescaledb_internal.data_node_hypertable_info(
+                CASE WHEN dht.node_up THEN
+                    dht.node_name
+                ELSE
+                    NULL
+                END, schema_name_in, dht.hypertable_name) x ON true
+            GROUP BY dht.hypertable_name
+        $function$
+        LANGUAGE sql
+        STRICT STABLE 
+        SECURITY DEFINER
+        --search path must be set for security definer
+        SET search_path = pg_temp;
+        REVOKE ALL ON FUNCTION SCHEMA_CATALOG.hypertable_remote_size(name) FROM PUBLIC;
+        GRANT EXECUTE ON FUNCTION SCHEMA_CATALOG.hypertable_remote_size(name) to prom_reader;
+    END IF;
+END;
+$block$
+;
+
+DO $block$
+BEGIN
+    IF SCHEMA_CATALOG.is_timescaledb_installed() AND SCHEMA_CATALOG.get_timescale_major_version() >= 2 THEN
+        -- two columns in _timescaledb_internal.compressed_chunk_stats were renamed in version 2.2
+        -- schema_name -> hypertable_schema
+        -- table_name -> hypertable_name
+        -- we explicit aliases to deal with this
```

Comment:
I"m not sure I understand this line? perhaps you could state a different way? or maybe there's a word missing?

Reply:
good catch. I did leave out a word. I'll fix and expand on it a bit